internal/api/verify.go

```diff
@@ -152,12 +154,11 @@ func (a *API) verifyGet(w http.ResponseWriter, r *http.Request, params *VerifyPa
 		case mail.EmailChangeVerification:
 			user, terr = a.emailChangeVerify(r, tx, params, user)
 			if user == nil && terr == nil {
-				// when double confirmation is required
-				rurl, err := a.prepRedirectURL(singleConfirmationAccepted, params.RedirectTo, flowType)
-				if err != nil {
-					return err
+				// only one OTP is confirmed at this point, so we return early and ask the user to confirm the second OTP
+				rurl, terr = a.prepRedirectURL(singleConfirmationAccepted, params.RedirectTo, flowType)
+				if terr != nil {
+					return terr
```

Comment:
Oh ok so to check that I'm understanding this correctly, before the diff the double redirect occurs when we have

line 160 (in code before change) which does a redirect and returns nil to the txn

```
http.Redirect(w, r, rurl, http.StatusSeeOther)	
return nil
```


which in turn goes to  line 220 (in code before change) and does a redirect which overwrites the `Location` error initially set by the Redirect on line 160?

Reply:
yeah and because the `timeoutResponseWriter` doesn't actually write the header to the actual response and writes to a buffer instead, it gets overwritten the second time since the key ('Location') is the same